### PR TITLE
tests: Fixes DNS tests for Windows

### LINKED ISSUE #81080: DNS tests fail on Windows due to getent usage
**Which jobs are failing**:

    sig-windows#flannel-overlay-windows-master

**Which test(s) are failing**:

    [sig-network] DNS should provide DNS for the cluster [Conformance]
    [sig-network] DNS should provide DNS for services [Conformance]

**Since when has it been failing**:

Since this PR merged: https://github.com/kubernetes/kubernetes/pull/75243

**Testgrid link**:

    https://testgrid.k8s.io/sig-windows#flannel-overlay-windows-master

**Reason for failure**:

The mentioned PR replaces the usage of ``hostname -i`` with ``getent hosts``, command which has issues on Windows, especially since the ``hosts`` file is not managed by Kubernetes.

**Anything else we need to know**:

/sig testing
/sig windows
/area conformance

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind failing-test

/sig testing
/sig windows
/area conformance

**What this PR does / why we need it**:

A previous commit replaced the usage of ``hostname -i`` with ``getent hosts`` in DNS probe commands (for testing IPv6), which breaks the Windows tests (``getent hosts`` is a Linux-specific command that looks into the ``/etc/hosts`` file, which, on Windows, is not managed by Kubelet).

This commit addresses this issue by executing ``getent hosts`` on IPv6 clusters and ``hostname -i`` on IPv4 clusters (Windows does not support IPv6 at this moment).

**Which issue(s) this PR fixes**:

Fixes #81080

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/network/dns_common.go [hunk 1]
```diff
@@ -474,7 +474,12 @@ func createProbeCommand(namesToResolve []string, hostEntries []string, ptrLookup
 
 	podARecByUDPFileName := fmt.Sprintf("%s_udp@PodARecord", fileNamePrefix)
 	podARecByTCPFileName := fmt.Sprintf("%s_tcp@PodARecord", fileNamePrefix)
-	probeCmd += fmt.Sprintf(`podARec=$$(getent hosts $$(hostname | awk '{print $1}') | tr ":." "-" | awk '{print $$1".%s.pod.%s"}');`, namespace, dnsDomain)
+
+	if framework.TestContext.IPFamily == "ipv4" {
+		probeCmd += fmt.Sprintf(`podARec=$$(hostname -i| awk -F. '{print $$1"-"$$2"-"$$3"-"$$4".%s.pod.%s"}');`, namespace, dnsDomain)
+	} else {
```
Comment (aojea) [inline test/e2e/network/dns_common.go]: kubernetes gives preference to IPv4, can we do the comparison the other way around? 
```
if framework.TestContext.IPFamily == "ipv6"` {
probeCmd ... ipv6
} else {
probeCmd ... ipv4
}

Comment (bclau) [reply test/e2e/network/dns_common.go]: Done

**Pull request merged**